Sources/DefaultsController.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+enum UserDefaultsState: Equatable {
+    case standard
+    case appGroup(String)
+
+    static func makeState(appGroupID: String?) -> UserDefaultsState {
+        if let appGroupID = appGroupID {
+            return .appGroup(appGroupID)
+        } else {
+            return .standard
+        }
+    }
+
+    static func == (lhs: UserDefaultsState, rhs: UserDefaultsState) -> Bool {
+        switch (lhs, rhs) {
+        case (.standard, .standard):
+            return true
+        case (.appGroup(let group1), .appGroup(let group2)):
+            return group1 == group2
+        default:
+            return false
+        }
+    }
+}
+
+struct DefaultsController {
+    private static var userDefaults: UserDefaults = .standard
+    private static let kDefaultsKey = "KronosStableTime"
+
+    private static var userDefaultsState: UserDefaultsState = .standard {
+        didSet {
+            guard oldValue != self.userDefaultsState else {
+                return
+            }
+
+            switch self.userDefaultsState {
+            case .standard:
+                self.userDefaults = .standard
+            case .appGroup(let groupName):
+                self.userDefaults = UserDefaults(suiteName: groupName) ?? .standard
+            }
+        }
+    }
+
+    static var stableTime: TimeFreeze? {
```

Comment:
I wouldn't create a computed property for this since we'll have to pay the cost of querying the storage on every access

Reply:
The querying process hasn’t changed from the existing version to the new one, since the defaults are only ever accessed from `sync()`. Do we want to change that?

Reply:
I went ahead and added caching of the current state if there is one so we don’t hit defaults on subsequent calls